### DISCUSSION TITLE
How should PKHeX deal with Hayley's barned Pokémon?

### DISCUSSION THREAD

**Discussion by Zazsona** [kwsch/PKHeX#4042] [Q&A] [answered]
'Ello! 'Ello!

For a bit of context, when you've got enough Pokémon in the Ranch that the total number is greater than 1,000, Hayley will offer to put her Pokémon in the barn, allowing you to store 1,000 of just your own Pokémon.

But those Pokémon don't get deleted from existence! It looks like Hayley's Pokémon basically just start occupying slots 1,001 onwards, which simply don't get rendered in game, but are still acknowledged in any counts:
![image](https://github.com/kwsch/PKHeX/assets/40435178/c89cd410-c950-42cd-890e-1662ac06fba0)

So, the question is, should we display these slots considering they would never normally be visible, accessible, or modifiable during regular gameplay? They nonetheless have Pokémon data in them, and one or two of the two people in the world who play Ranch might find it handy!

I've not explored some specific scenarios, like what the actual maximum is, whether the Platinum Update has any changes to this behaviour, and what the implications are of storing a player-owned Pokémon in those slots are, but figured I'd start this discussion for visibility & opinions before I get distracted by some other esoteric Ranch behaviour!

**Comment by kwsch** [comment, marked as answer]
This feels similar to rental teams in stadium saves, where the data really shouldn't be mutable but can still be viewed. There isn't really a good abstraction in the GUI for an immutable shadow-box besides the `SAV_GroupViewer` that is called by `SAVEditor.B_OtherSlots_Click`.

Reusing that logic should get you the end result you're looking for.
